src/engine/InputQuery.cpp

```diff
@@ -590,15 +590,24 @@ bool InputQuery::constructNetworkLevelReasoner()
     List<unsigned> inputs = getInputVariables();
     nlr->addLayer( 0, NLR::Layer::INPUT, inputs.size() );
     unsigned index = 0;
+
+    auto *layer = nlr->getLayerToUpdate( 0 );
```

Comment:
Change from "auto *layer" to Layer *inputLayer

We try to use auto only in for loops
And informative names are better